server/src/server_config.rs

```
@@ -0,0 +1,312 @@
+// Copyright (c) 2025 Elektrobit Automotive GmbH
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License, Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::Arguments;
+use common::DEFAULT_SOCKET_ADDRESS;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::net::SocketAddr;
+use toml::from_str;
+
+const CONFIG_VERSION: &str = "v1";
+pub const DEFAULT_SERVER_CONFIG_FILE_PATH: &str = "/etc/ankaios/ank-server.conf";
+
+#[cfg(not(test))]
+use std::fs::read_to_string;
+
+#[cfg(test)]
+fn read_to_string(file_path_content: &str) -> std::io::Result<String> {
+    Ok(file_path_content.to_string())
+}
+
+pub fn get_default_address() -> Option<SocketAddr> {
+    DEFAULT_SOCKET_ADDRESS.parse().ok()
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum ConversionErrors {
+    WrongVersion(String),
+    ConflictingCertificates(String),
+}
+
+fn convert_to_socket_address<'de, D>(deserializer: D) -> Result<Option<SocketAddr>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+
+    Ok(s.parse::<SocketAddr>().ok())
```

So if I cannot parse it, it would return None. It would at least not use the default, but the code in main.rs on line 144 panics because of the unwrap().
We should propagate the error here and not convert to option.